browser/ui/browser_commands.cc

```diff
@@ -898,6 +903,56 @@ void ScrollTabToBottom(Browser* browser) {
   contents->ScrollToBottomOfDocument();
 }
 
+namespace {
+
+/**
+ * @class BookmarksExportListener
+ * @brief A listener class for handling bookmark export file selection.
+ *
+ * This class is responsible for showing a file dialog to the user for selecting
+ * the location to save exported bookmarks.
+ *
+ * @note The lifetime of this class is tied to the FileSelected dialog. It will
+ * be automatically deleted when the dialog is closed, a file is selected, or
+ * the dialog is cancelled.
+ */
+class BookmarksExportListener : public ui::SelectFileDialog::Listener {
+ public:
+  explicit BookmarksExportListener(Profile* profile)
+      : profile_(profile),
+        file_selector_(ui::SelectFileDialog::Create(this, nullptr)) {}
+  void FileSelected(const ui::SelectedFileInfo& file, int index) override {
+    bookmark_html_writer::WriteBookmarks(profile_, file.file_path, nullptr);
+    delete this;
+  }
+  void ShowFileDialog(Browser* browser) {
+    const std::string exported_bookmarks_filename =
+        base::UnlocalizedTimeFormatWithPattern(base::Time::Now(), "yyyy_MM_dd",
+                                               nullptr) +
+        "_brave_browser_bookmarks.html";
```

Comment:
should this name be localized?

Reply:
Looks like there already is one!

I think we should use `IDS_EXPORT_BOOKMARKS_DEFAULT_FILENAME`

Reply:
Let's replace with that - nice find! 😄